pyvortex/python/vortex/dataset.py

```diff
@@ -0,0 +1,299 @@
+        use_threads: bool | None = None,
+        memory_pool: pa.MemoryPool = None,
+    ) -> int:
+        """Not implemented."""
+        raise NotImplementedError("count_rows")
+
+    def filter(self, expression: pc.Expression) -> "VortexDataset":
+        """Not implemented."""
+        raise NotImplementedError("filter")
+
+    def get_fragments(self, filter: pc.Expression | None = None) -> Iterator[pa.dataset.Fragment]:
+        """Not implemented."""
+        raise NotImplementedError("get_fragments")
+
+    def head(
+        self,
+        num_rows: int,
+        columns: list[str] | None = None,
+        filter: pc.Expression | None = None,
+        batch_size: int | None = None,
+        batch_readahead: int | None = None,
+        fragment_readahead: int | None = None,
+        fragment_scan_options: pa.dataset.FragmentScanOptions | None = None,
+        use_threads: bool | None = None,
+        memory_pool: pa.MemoryPool = None,
+    ) -> pa.Table:
+        if batch_readahead is not None:
+            raise ValueError("batch_readahead not supported")
+        if fragment_readahead is not None:
+            raise ValueError("fragment_readahead not supported")
+        if fragment_scan_options is not None:
+            raise ValueError("fragment_scan_options not supported")
+        if use_threads is True:
+            warnings.warn("Vortex does not support threading. Ignoring use_threads=True")
+        if columns is not None and len(columns) == 0:
+            raise ValueError("empty projections are not currently supported")
+        del memory_pool
+        if filter is not None:
+            filter = arrow_to_vortex_expr(filter, self.schema)
+        return self._dataset.to_array(columns, batch_size, filter).slice(0, num_rows).to_arrow_table()
+
+    def join(
+        self,
+        right_dataset,
+        keys,
+        right_keys=None,
+        join_type=None,
+        left_suffix=None,
+        right_suffix=None,
+        coalesce_keys=True,
+        use_threads: bool | None = None,
+    ) -> pa.dataset.InMemoryDataset:
+        """Not implemented."""
+        raise NotImplementedError("join")
+
+    def join_asof(self, right_dataset, on, by, tolerance, right_on=None, right_by=None) -> pa.dataset.InMemoryDataset:
+        """Not implemented."""
+        raise NotImplementedError("join_asof")
+
+    def replace_schema(self, schema: pa.Schema):
+        """Not implemented."""
+        raise NotImplementedError("replace_schema")
+
+    def scanner(
+        self,
+        columns: list[str] | None = None,
+        filter: pc.Expression | None = None,
+        batch_size: int | None = None,
+        batch_readahead: int | None = None,
+        fragment_readahead: int | None = None,
+        fragment_scan_options: pa.dataset.FragmentScanOptions | None = None,
+        use_threads: bool | None = None,
+        memory_pool: pa.MemoryPool = None,
+    ) -> pa.dataset.Scanner:
+        """Not implemented."""
```

Comment:
Is this truly not implemented or leftover?

Reply:
Indeed a mistake; already fixed in develop: https://github.com/spiraldb/vortex/commit/54c19938d1b2dcff5bdcdcca673f353f9a9d9610#diff-482b5dfc27f9f86d89ce96b259d2781464eb749b777c79fe2dca8231e6f26983L117